odl/tomo/backends/astra_cpu.py

```diff
@@ -0,0 +1,192 @@
+    algo_id = astra_algorithm('forward', ndim, vol_id, sino_id, proj_id,
+                              impl='cpu')
+
+    # Run algorithm and delete it
+    astra.algorithm.run(algo_id)
+
+    # Flip detector pixels for fanflat
+    if isinstance(geometry, FanFlatGeometry):
+        out[:] = out.asarray()[::-1, ::-1]
+
+    # Delete ASTRA objects
+    astra_cleanup()
+
+    return out
+
+
+def astra_cpu_backward_projector_call(proj_data, geometry, reco_space,
+                                      out=None):
+    """Run an ASTRA backward projection on the given data using the CPU.
+
+    Parameters
+    ----------
+    proj_data : `DiscreteLpVector`
+        Projection data to which the backward projector is applied
+    geometry : `Geometry`
+        Geometry defining the tomographic setup
+    reco_space : `DiscreteLp`
+        Space to which the calling operator maps
+    out : `DiscreteLpVector` or `None`, optional
+        Vector in the reconstruction space to which the result is written.
+        Default: `None`
+
+    Returns
+    -------
+    out : ``reco_space`` element
+        Reconstruction data resulting from the application of the backward
+        projector
+    """
+    if not isinstance(proj_data, DiscreteLpVector):
+        raise TypeError('projection data {!r} is not a `DiscreteLpVector` '
+                        'instance.'.format(proj_data))
+    if not isinstance(geometry, Geometry):
+        raise TypeError('geometry  {!r} is not a `Geometry` instance.'
+                        ''.format(geometry))
+    if not isinstance(reco_space, DiscreteLp):
+        raise TypeError('reconstruction space {!r} is not a `DiscreteLp` '
+                        'instance.'.format(reco_space))
+
+    if reco_space.grid.ndim != geometry.ndim:
+        raise ValueError('dimensions {} of reconstruction space and {} of '
+                         'geometry do not match.'.format(reco_space.grid.ndim,
+                                                         geometry.ndim))
+
+    ndim = proj_data.space.grid.ndim
+
+    # Create astra geometries
+    vol_geom = astra_volume_geometry(reco_space)
+    proj_geom = astra_projection_geometry(geometry)
+
+    if out is None:
+        out = reco_space.element()
+
+    # Create ASTRA data structures
+    vol_id = astra_data(vol_geom, datatype='volume', data=out,
+                        ndim=reco_space.grid.ndim)
+    sino_id = astra_data(proj_geom, datatype='projection', data=proj_data)
+
+    # Create projector
+    vol_interp = proj_data.space.interp
+    proj_id = astra_projector(vol_interp, vol_geom, proj_geom, ndim,
+                              impl='cpu')
+
+    # Create algorithm
+    algo_id = astra_algorithm('backward', ndim, vol_id, sino_id, proj_id,
+                              impl='cpu')
+
+    # Run algorithm and delete it
+    astra.algorithm.run(algo_id)
+
+    # flip both dimensions = rotate 180 degrees
+    # TODO: Maybe flipping angles in the geometry would be better
```

Comment:
For the record, flipping the angles in the geometry is probably the only fully correct way to work around this inconsistency in astra at the moment. Flipping the data like this works as long as your image and detector are centered around 0, which they generally of course will be.